common/test/test_migrations/utils.py

```diff
@@ -0,0 +1,38 @@
+""" General class for Test migrations. """
+from django.apps import apps
+from django.test import TransactionTestCase
+from django.db.migrations.executor import MigrationExecutor
+from django.db import connection
+
+
+class TestMigrations(TransactionTestCase):
+    """ Base class for testing migrations. """
+    migrate_from = None
+    migrate_to = None
+    app = None
+
+    def setUp(self):
+        assert self.migrate_from and self.migrate_to, \
+            "TestCase '{}' must define migrate_from and migrate_to properties".format(type(self).__name__)
+        assert self.app, "app must be define in the TestCase"
+        self.migrate_from = [(self.app, self.migrate_from)]
+        self.migrate_to = [(self.app, self.migrate_to)]
+        executor = MigrationExecutor(connection)
+        old_apps = executor.loader.project_state(self.migrate_from).apps
+
+        # Reverse to the original migration
+        executor.migrate(self.migrate_from)
+
+        self.setUpBeforeMigration(old_apps)
+
+        # Run the migration to test
+        executor.migrate(self.migrate_to)
+
+        self.apps = executor.loader.project_state(self.migrate_to).apps
+
+    def setUpBeforeMigration(self, apps):
+        """
+        Will run before befor migration using config field migrate_from.
+        Implemented in drived class.
+        """
+        pass
```

Comment:
can we add a method here called `perform_reverse_migration` that does something like:

```python
def perform_reverse_migration(self):
    executor = MigrationExecutor(connection)
    executor.migrate(self.migrate_from)
    self.apps = executor.loader.project_state(self.migrate_from).apps
```

That way we can easily test that reverse migrations do what we expect.